hello/hello_child_workflow.py

```diff
@@ -0,0 +1,57 @@
+import asyncio
+from dataclasses import dataclass
+
+from temporalio import workflow
+from temporalio.client import Client
+from temporalio.worker import Worker
+
+
+@dataclass
+class ComposeGreetingInput:
+    greeting: str
+    name: str
+
+
+@workflow.defn
+class ComposeGreeting:
+    @workflow.run
+    async def run(self, input: ComposeGreetingInput) -> str:
+        return f"{input.greeting}, {input.name}!"
+
+
+@workflow.defn
+class GreetingWorkflow:
+    @workflow.run
+    async def run(self, name: str) -> str:
+        return await workflow.execute_child_workflow(
+            ComposeGreeting.run,
+            ComposeGreetingInput("Hello", name),
+            id="hello-child-workflow-workflow-child-id",
+        )
+
+
+async def main():
+    # Start client
+    client = await Client.connect("http://localhost:7233")
+
+    # Run a worker for the workflow
+    async with Worker(
+        client,
+        task_queue="hello-child-workflow-task-queue",
+        workflows=[GreetingWorkflow, ComposeGreeting],
```

Comment:
Looking at all of this registration logic and I'm pretty sure Python users will find this annoying.
So many Python frameworks out there collect just by using the decorator.
We should consider adding this ability to the SDK IMHO.

Reply:
This would be a user defined global.
Something along these lines.


```python
collection = Collection()

@collection.workflow.defn
...

@collection.activity.defn
...

# or

@workflow.defn(collections=[collection])
...

Worker(**collection.worker_config)
```

This roughly follows a common pattern that popular frameworks like celery, fastapi and flask use.

Reply:
Sounds just as easy to create your own decorator instead of having our decorator do this for you. This is especially true with class and method-based activities that need an instance to be created by the user to be registered anyways.